lib/core/Axios.js

```diff
@@ -45,20 +47,62 @@ Axios.prototype.request = function request(config) {
     config.method = 'get';
   }
 
-  // Hook up interceptors middleware
-  var chain = [dispatchRequest, undefined];
-  var promise = Promise.resolve(config);
+  var requestCancelled = config.cancelToken && config.cancelToken.reason;
 
+  // filter out skipped interceptors
+  var requestInterceptorChain = [];
+  var synchronousChain = [];
   this.interceptors.request.forEach(function unshiftRequestInterceptors(interceptor) {
-    chain.unshift(interceptor.fulfilled, interceptor.rejected);
+    if (typeof interceptor.runWhen === 'function' && interceptor.runWhen(config) === false) {
```

Comment:
I chose to skip for `false` only to make sure that the user is explicit when they want the interceptors to be skipped (I felt its safer to just run the interceptor if a falsy value is returned because the intent is not clear).

Reply:
Fine. Just remember to emphasize that in the document.